services/analyzer/sequelize-tables.js

```diff
@@ -0,0 +1,307 @@
+              foreignKey,
+              foreignKey.isPrimary || foreignKey.isUnique ? 'hasOne' : 'hasMany',
+            ),
+          );
+        }
+      }
+    });
+  });
+}
+
+// NOTICE: Set the remaining fields
+async function createTableSchema(schema, foreignKeys, primaryKeys, references, table) {
+  const fields = [];
+  await P.each(Object.keys(schema), async (nameColumn) => {
+    const columnInfo = schema[nameColumn];
+    const type = await columnTypeGetter.perform(columnInfo, nameColumn, table);
+    const foreignKey = _.find(foreignKeys, { columnName: nameColumn, columnType: 'FOREIGN KEY' });
+
+    if ((!foreignKey
+          || !foreignKey.foreignTableName
+          || !foreignKey.columnName || columnInfo.primaryKey) && type) {
+      // NOTICE: If the column is of integer type, named "id" and primary, Sequelize will
+      //         handle it automatically without necessary declaration.
+      if (nameColumn !== 'id' || type !== 'INTEGER' || !columnInfo.primaryKey) {
+        // NOTICE: Handle bit(1) to boolean conversion
+        if (columnInfo.defaultValue === "b'1'" || columnInfo.defaultValue === '((1))') { columnInfo.defaultValue = true; }
+        if (columnInfo.defaultValue === "b'0'" || columnInfo.defaultValue === '((0))') { columnInfo.defaultValue = false; }
+
+        const field = {
+          name: _.camelCase(nameColumn),
+          nameColumn,
+          type,
+          primaryKey: columnInfo.primaryKey,
+          defaultValue: columnInfo.defaultValue,
+        };
+
+        fields.push(field);
+      }
+    }
+  });
+
+  const options = {
+    underscored: isUnderscored(fields),
+    timestamps: hasTimestamps(fields),
+    hasIdColumn: hasIdColumn(fields, primaryKeys),
+    hasPrimaryKeys: !_.isEmpty(primaryKeys),
+  };
+
+  return {
+    fields,
+    references,
+    primaryKeys,
+    options,
+  };
+}
+
+async function analyzeSequelizeTables(databaseConnection, config, allowWarning) {
+  const lianaSchema = {};
+
+  queryInterface = databaseConnection.getQueryInterface();
+  tableForeignKeysAnalyzer = new TableForeignKeysAnalyzer(databaseConnection, config.dbSchema);
+
+  columnTypeGetter = new ColumnTypeGetter(databaseConnection, config.dbSchema || 'public', allowWarning);
+
+  if (config.dbSchema) {
+    const schemaExists = await queryInterface.sequelize
+      .query(
+        'SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = ?;',
+        { type: queryInterface.sequelize.QueryTypes.SELECT, replacements: [config.dbSchema] },
+      )
+      .then((result) => !!result.length);
+
+    if (!schemaExists) {
+      const message = 'This schema does not exists.';
+      return terminate(1, {
+        errorCode: 'database_authentication_error',
+        errorMessage: message,
+        logs: [message],
+      });
+    }
+  }
+
+  // Build the db schema.
+  const databaseSchema = {};
+  await P.mapSeries(showAllTables(databaseConnection, config.dbSchema), async (table) => {
```

Comment:
You are doing multiple times the showAllTables Oo ?! Do it once then reuse that variable